lib/evaluators/AggregateEvaluator.ts

```diff
@@ -313,3 +312,51 @@ function extractNumericValueAndTypeOrError(term: RDF.Term): { value: number, typ
   const value = parseXSDFloat(term.value);
   return { type, value };
 }
+
+function extractValue(term: RDF.Term): {value: any, type:string}  {
+  if (term.termType !== 'Literal') {
+    throw new Error('Term is not a literal');
+  }
+  const transformedLit = transformLiteral(term);
+  return {type: transformedLit.typeURL.value, value: transformedLit.typedValue};
+}
+
+function getExtreme(state: ExtremeState[], isMin:boolean): RDF.Term {
+  // Check if all elements are of the same type
+  let sameType = true;
+  const firstElem = extractValue(state[0].term);
+  const type = firstElem.type;
+  for (const element of state) {
+    const extractedElem = extractValue(element.term);
+    if (extractedElem.type !== type) {
+      sameType = false;
+      break;
+    }
+  }
+
+  // Different types will be compared as strings
+  let extreme = state[0].term;
+  let extremeValue;
+  if (sameType) {
```

Comment:
`value` is always a string, right? So this if-else statement should not be needed.

Reply:
The term is retrieved from extractValue(term: RDF.Term). Which gets the value by using transformLiteral(term).value. This function returns values from types different from string such as dates, numericalLiterals... https://github.com/comunica/sparqlee/blob/master/lib/Transformation.ts#L73 So I don't think it is always a string in this case.